From 66ea3effe607b63f0e3c5c2ee05fda68a8216397 Mon Sep 17 00:00:00 2001
From: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
Date: Tue, 25 Feb 2025 15:44:04 +0800
Subject: [PATCH] Revert "PCI: imx6: Skip controller_id generation logic for
 i.MX7D"

This reverts commit 2a12efc567a270a155e3b886258297abd79cdea0.

Commit 2a12efc567a ("PCI: imx6: Skip controller_id generation logic for
i.MX7D") inadvertently removed the pciephy_reset and apps_reset logic,
which caused issues with PCIe reset handling on i.MX7D boards, so revert
it.

Upstream-Status: Inappropriate [native]

Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index dbb4919054d6..93e2a7b44256 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1856,6 +1856,7 @@ static int imx6_pcie_probe(struct platform_device *pdev)
 	switch (imx6_pcie->drvdata->variant) {
 	case IMX8MQ:
 	case IMX8MQ_EP:
+	case IMX7D:
 		if (dbi_base->start == IMX8MQ_PCIE2_BASE_ADDR)
 			imx6_pcie->controller_id = 1;
 
-- 
2.46.2

